gravitee-node-monitoring/src/main/java/io/gravitee/node/monitoring/healthcheck/NodeHealthCheckManagementEndpoint.java

```diff
@@ -57,31 +58,35 @@ public String path() {
 
     @Override
     public void handle(RoutingContext ctx) {
+        Set<String> probeIds;
+        if (ctx.queryParams().contains(PROBE_FILTER)) {
+            probeIds =
+                ctx.queryParams().getAll(PROBE_FILTER).stream().flatMap(s -> Arrays.stream(s.split(","))).collect(Collectors.toSet());
+        } else {
+            probeIds = null;
+        }
         probeEvaluator
-            .evaluate()
+            .evaluate(probeIds)
             .thenAccept(probeResults -> {
-                final Map<Probe, Result> probes = probeResults
+                final Map<String, Result> results = probeResults
                     .entrySet()
                     .stream()
-                    .filter(entry ->
-                        ctx.queryParams().contains(PROBE_FILTER)
-                            ? ctx.queryParams().get(PROBE_FILTER).contains(entry.getKey().id())
-                            : entry.getKey().isVisibleByDefault()
-                    )
-                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+                    .filter(entry -> probeIds != null ? probeIds.contains(entry.getKey().id()) : entry.getKey().isVisibleByDefault())
+                    .collect(Collectors.toMap(entry -> entry.getKey().id(), Map.Entry::getValue));
 
-                boolean healthyProbe = probes.values().stream().allMatch(Result::isHealthy);
+                if (probeIds != null) {
+                    probeIds
+                        .stream()
+                        .filter(probeId -> !results.containsKey(probeId))
+                        .forEach(probeId -> results.put(probeId, Result.unhealthy("probe not found")));
+                }
 
                 HttpServerResponse response = ctx.response();
+                boolean healthyProbe = results.values().stream().allMatch(Result::isHealthy);
                 response.setStatusCode(healthyProbe ? HttpStatusCode.OK_200 : HttpStatusCode.INTERNAL_SERVER_ERROR_500);
```

Comment:
In case of unknown probes, shouldn't be a 404 error rather than a 500 ?

Reply:
The result can contain existing probes healthy or unhealthy in addition to unknown probes, so 404 won't make it for those cases. It is easier to stick with 500.